pkg/cmd/roachtest/tests/mixed_version_cdc.go

```diff
@@ -0,0 +1,318 @@
+// Copyright 2022 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package tests
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/cdctest"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/option"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/registry"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test"
+	"github.com/cockroachdb/cockroach/pkg/testutils/sqlutils"
+	"github.com/cockroachdb/cockroach/pkg/util/retry"
+	"github.com/cockroachdb/cockroach/pkg/util/timeutil"
+	"github.com/cockroachdb/cockroach/pkg/util/version"
+)
+
+const (
+	// how many resolved timestamps to wait for before considering the
+	// system to be working as intended.
+	requestedResolved = 20
```

Comment:
What's the significance of `20`?

Reply:
Chosen empirically based on how long it takes to reach 20 resolved timestamps. With `20`, the test typically spends non-negligible time in all stages of the test. It's a little arbitrary, but that's how other CDC tests operate and I was trying to use something that is known to work here. Open to changing if you have suggestions.

Reply:
This seems very reasonable. I wonder if we should also set the minimum duration between subsequent "resolved timestamps" events, e.g., `resolved='30s'`. The rationale is to prevent the case where each resolved timestamp corresponds to exactly one change; i.e., ideally we want to test more than 20 changes in total.

Reply:
Not sure yet about the use of this; using resolved option by itself, sometimes hides issues -- (it forces to emit resolved message and kinda acts as a force flush function)...  (Just typing this in since I do top-bottom review).